### PR TITLE
SITES-1318 Filter out non-existent permissions.

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Filter permissions before adding them

# Needed By (Date)
- :man_shrugging: 

# Urgency
- low-ish

# Steps to Test
1. use this branch on military
1. run db updates
1. verify updates complete without errors.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install [hunk 1]
```diff
@@ -216,5 +216,10 @@ function stanford_jumpstart_vpsa_permissions_update_7401_get_permissions() {
     'module' => 'node',
   );
 
-  return $permissions;
+  $permissions_modules = user_permission_get_modules();
+  // If a permission doesn't exist, filter it out to prevent errors when trying
+  // to add a non-existent permission.
+  return array_filter($permissions, function ($permission) use ($permissions_modules) {
```
Comment (jbickar) [inline modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install:222]: Am I correct that this is an anonymous function? `diseval` might puke on it if so.

**Comment by jbickar** [comment]
This fixes the upgrade path on `sara`, `samarkazupgrade`, and `samilitaryupgrade`.

File: modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install (hunk 1 above)
Comment (pookmish) [reply modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install:222]: Theres several modules that do a similar thing. including Guzzle in capx. Hopefully diseval doesn't hate it https://github.com/SU-SWS/stanford_capx/blob/7.x-3.x/includes/vendor/guzzlehttp/psr7/src/Uri.php#L634

Comment (pookmish) [reply modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install:222]: and its used alot in D8.

Comment (sherakama) [reply modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install:222]: FWIK Anon functions are ok. `create_function` is not.

https://github.com/mk-j/PHP_diseval_extension

**Review by sherakama — APPROVED** [review]

File: modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install (hunk 1 above)
Comment (jbickar) [reply modules/stanford_jumpstart_vpsa_permissions/stanford_jumpstart_vpsa_permissions.install:222]: Ah that's right. My memory was fuzzy.

**Review by jbickar — APPROVED** [review]
Thank you both, @pookmish and @sherakama.

**Pull request merged**